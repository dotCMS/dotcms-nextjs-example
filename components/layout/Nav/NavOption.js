import React from 'react';
import PropTypes from 'prop-types';
import RouterLink from '../../Shared/RouterLink';

const NavOption = ({ item }) => {
    return (
        <RouterLink key={item.folder} className="rd-nav-link" pathname={item.href}>
            {item.title}
        </RouterLink>
    );
};

NavOption.propTypes = {
    item: PropTypes.shape({
        folder: PropTypes.string,
        href: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired
    }).isRequired
};

export default NavOption;
