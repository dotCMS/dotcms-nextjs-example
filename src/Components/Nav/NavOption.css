.nav-option {
    font-size: 0.75rem;
    font-weight: 500;
    letter-spacing: 2px;
}

.nav-option,
.nav-option:hover {
    color: var(--white);
    text-transform: uppercase;
}

.nav-menu__links .nav-option.nav-link.active {
    background-color: transparent;
    color: rgba(255, 255, 255, 0.5);
}
