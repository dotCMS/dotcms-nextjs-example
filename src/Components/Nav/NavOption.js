import React from 'react';
import PropTypes from 'prop-types';
import { NavItem, NavLink } from 'reactstrap';
import { Link } from "react-router-dom";

const NavOption = ({ item }) => {
    return (
        <NavItem key={item.folder}>
            <NavLink tag={Link} active={item.href === window.location.pathname} to={item.href}>
                {item.title}
            </NavLink>
        </NavItem>
    );
};

NavItem.propTypes = {
    item: PropTypes.object.isRequired
};

export default NavOption;
