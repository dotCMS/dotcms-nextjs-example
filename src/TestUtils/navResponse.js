export const NavResponse = {
    entity: {
        children: [
            {
                href: '/about',
                title: 'About Us',
                folder: '123',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/products',
                title: 'Products',
                folder: '456',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/resources',
                title: 'Resources',
                folder: '789',
                type: 'folder',
                children: [
                    {
                        href: '/videos',
                        title: 'Videos',
                        type: 'htmlpage',
                        folder: null
                    },
                    {
                        href: '/blog',
                        title: 'Blog',
                        type: 'htmlpage',
                        folder: null
                    }
                ]
            }
        ]
    }
};
