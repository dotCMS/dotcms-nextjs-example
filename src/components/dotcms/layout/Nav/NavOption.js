import React from 'react';
import PropTypes from 'prop-types';
import RouterLink from '../../RouterLink';

const NavOption = ({ item }) => {
    return (
        <RouterLink className="rd-nav-link" href={item.href} key={item.folder}>
            {item.title}
        </RouterLink>
    );
};

NavOption.propTypes = {
    item: PropTypes.shape({
        folder: PropTypes.string,
        href: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired
    }).isRequired
};

export default NavOption;
