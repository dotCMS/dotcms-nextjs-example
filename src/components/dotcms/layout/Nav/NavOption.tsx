// Dependencies
import * as React from 'react'

// Internals
import { Link } from '@/components'

export type NavOptionProps = {
  item: {
    folder: string
    href: string
    title: string
  }
}

export const NavOption = ({ item }: NavOptionProps) => {
  return (
    <Link className="rd-nav-link" href={item.href} key={item.folder}>
      {item.title}
    </Link>
  )
}

export default NavOption
