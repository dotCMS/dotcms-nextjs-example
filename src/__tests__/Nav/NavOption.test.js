import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NavOption from '../../Components/Nav/NavOption';
import { NavResponse } from '../../TestUtils/navResponse';

describe('<NavOption />', () => {
    it('renders NavOption correctly', () => {
        const wrapper = shallow(<NavOption item={NavResponse.entity.children[0]} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
