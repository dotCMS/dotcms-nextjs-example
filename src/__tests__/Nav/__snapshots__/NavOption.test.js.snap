// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavOption /> renders NavOption correctly 1`] = `
<NavItem
  key="456"
  tag="li"
>
  <NavLink
    active={false}
    tag={[Function]}
    to="/products"
  >
    Products
  </NavLink>
</NavItem>
`;
